### ISSUE TITLE
💅  False positives for `noUnknownPseudoElement` after updating to 2.4.0

### ISSUE THREAD

**Issue by planttheidea** [biomejs/biome#9081]
### Environment information

<details>

```bash
CLI:
  Version:                      2.4.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.11.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.8.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useIsNan
    correctness/useParseIntRadix
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noDuplicateAttributes
    nursery/noNestedPromises
    nursery/noProto
    nursery/noScriptUrl
    nursery/useSpread
    performance/noAccumulatingSpread
    performance/noDynamicNamespaceImportAccess
    security/noBlankTarget
    security/noGlobalEval
    style/noDescendingSpecificity
    style/noNonNullAssertion
    style/useArrayLiterals
    style/useConst
    style/useDeprecatedReason
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useShorthandFunctionType
    style/useTemplate
    suspicious/noApproximativeNumericConstant
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0
```

</details>

### Rule name

noUnknownPseudoElement

### Playground link

https://next.biomejs.dev/playground/?lineWidth=100&indentStyle=space&quoteStyle=single&analyzerFixMode=safeAndUnsafeFixes&cssModules=true&code=KgAgAHsACgAgACAALQB3AGUAYgBrAGkAdAAtAHQAYQBwAC0AaABpAGcAaABsAGkAZwBoAHQALQBjAG8AbABvAHIAOgAgAHQAcgBhAG4AcwBwAGEAcgBlAG4AdAA7AAoAIAAgAGIAbwB4AC0AcwBpAHoAaQBuAGcAOgAgAGIAbwByAGQAZQByAC0AYgBvAHgAOwAKACAAIABjAG8AbABvAHIAOgAgAGkAbgBoAGUAcgBpAHQAOwAKACAAIABmAG8AbgB0AC0AdwBlAGkAZwBoAHQAOgAgAGkAbgBoAGUAcgBpAHQAOwAKACAAIABtAGEAcgBnAGkAbgA6ACAAMAA7AAoAIAAgAHAAYQBkAGQAaQBuAGcAOgAgADAAOwAKACAAIABwAG8AcwBpAHQAaQBvAG4AOgAgAHIAZQBsAGEAdABpAHYAZQA7AAoAIAAgAHYAZQByAHQAaQBjAGEAbAAtAGEAbABpAGcAbgA6ACAAbQBpAGQAZABsAGUAOwAKAAoAIAAgACYAOgA6AGEAZgB0AGUAcgAsAAoAIAAgACYAOgA6AGIAZQBmAG8AcgBlACAAewAKACAAIAAgACAAYQBsAGwAOgAgAHUAbgBzAGUAdAA7AAoAIAAgACAAIABiAG8AeAAtAHMAaQB6AGkAbgBnADoAIABpAG4AaABlAHIAaQB0ADsACgAgACAAfQAKAH0A&language=css&ruleDomains.react=recommended&ruleDomains.project=recommended&ruleDomains.test=recommended

### Expected result

After updating to `2.4.0`, I am getting false positives with the `noUnknownPseudoElement` lint rule on very standard ones like `::after`. After playing in the playground above, it appears only when I activate CSS Modules (which in my repo, I am). When I deactivate CSS Modules, the error disappears.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by carry0987** [comment]
# `noUnknownPseudoElement` falsely reports standard pseudo-elements as unknown

## Environment

- **Biome version:** 2.4.0
- **OS:** macOS

## Description

The `lint/correctness/noUnknownPseudoElement` rule incorrectly flags the following standard CSS pseudo-elements as unknown:

- `::after`
- `::before`
- `::backdrop`
- `::file-selector-button`

These are all well-established, standard pseudo-elements documented on MDN.

Ironically, the diagnostic message itself suggests using these exact pseudo-elements as replacements:

> Use a known pseudo-elements instead, such as: **after**, **backdrop**, **before**, etc.

This indicates that the known pseudo-elements list does include them, but the matching/lookup logic fails to correctly compare against the parsed selector.

## Input Code

```css
@layer base {
    *,
    ::after,
    ::before,
    ::backdrop,
    ::file-selector-button {
        border-color: var(--color-border, currentcolor);
    }
}
```

## Biome Configuration

```json
{
  "$schema": "https://biomejs.dev/schemas/2.4.0/schema.json",
  "css": {
    "parser": {
      "allowWrongLineComments": true,
      "cssModules": true,
      "tailwindDirectives": true
    },
    "linter": {
      "enabled": true
    }
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true
    }
  }
}
```

## Expected Behavior

No errors should be reported. `::after`, `::before`, `::backdrop`, and `::file-selector-button` are all valid CSS pseudo-elements and should be recognized by the rule.

## Actual Behavior

Four errors are emitted:

```
src/shared/styles/globals.css:73:7 lint/correctness/noUnknownPseudoElement
  × Unexpected unknown pseudo-elements: after

src/shared/styles/globals.css:74:7 lint/correctness/noUnknownPseudoElement
  × Unexpected unknown pseudo-elements: before

src/shared/styles/globals.css:75:7 lint/correctness/noUnknownPseudoElement
  × Unexpected unknown pseudo-elements: backdrop

src/shared/styles/globals.css:76:7 lint/correctness/noUnknownPseudoElement
  × Unexpected unknown pseudo-elements: file-selector-button
```

**Comment by PaulRBerg** [comment]
Can reproduce, getting a bunch of false positive errors like so:

````
core/isomorphic/theme/base.css:7:5 lint/correctness/noUnknownPseudoElement ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unexpected unknown pseudo-elements: before

    6 │   *,
  > 7 │   ::before,
      │     ^^^^^^
    8 │   ::after {
    9 │     box-sizing: border-box;

  ℹ See MDN web docs for more details.

  ℹ Use a known pseudo-elements instead, such as:

  - after
  - backdrop
  - before
  - etc.
````

**Comment by planttheidea** [comment]
Just updated to `2.4.2` and confirmed this is fixed. Thanks for the quick turnaround!
